app/controllers/baskets_controller.rb

```diff
@@ -0,0 +1,33 @@
+class BasketsController < ApplicationController
+  def index
+    @baskets = Basket.all
+  end
+
+  def show
+    @basket = Basket.find(params[:id])
+  end
+
+  def new
+    @basket = Basket.new
+  end
+
+  def create
+    @basket = build_goods(Basket.new(sales_taxes: 0, total: 0))
+    @basket.save
+    redirect_to basket_path(@basket.id)
+  end
+
+  private
+
+  def build_goods(basket)
```

Comment:
It's good that you used a private to not expose it as a public method. But this logic might not belong to a controller.

A controller is a type of class that's responsible for making sense out of a request made in an application and producing appropriate output. So this logic to split and parse a file can go in another class. That way, you can easily unit test that class and the controller just has one line to call that class.